ee/hogai/eval/ci/eval_memory_onboarding.py

```diff
@@ -0,0 +1,397 @@
+            name="has_correct_style",
+            prompt_template="""Determine if the scraped content follows the required formatting structure.
+
+A PASS requires ALL of the following:
+- Starts with exactly "__Here's what I found on [some product/domain name]:__" (bold formatting with colon)
+- Contains at least one section with #### heading format (h4 markdown headers)
+- Sections should include relevant topics like "Product features", "User/Customer segments", "Business model", "Technical details", or "Brief history"
+- Uses bullet points (-) or structured formatting within sections (not just paragraphs)
+- Content is well-organized with clear separation between topics
+- No follow-up suggestions or calls-to-action at the end
+- No citation anywhere (i.e. link in parentheses)
+
+A FAIL occurs if ANY of the following is true:
+- Missing or incorrect opening format (doesn't start with "__Here's what I found on...:__")
+- No #### section headings found
+- Poor organization with wall-of-text paragraphs instead of structured sections
+- Contains follow-up suggestions like "Would you like to know more?" or "Contact us"
+- Sections lack bullet points or clear structure
+- Generic formatting without the specified markdown structure
+- A citation link is found
+- There is any follow-up suggestion for the reader (like "Let me know if you'd like X" or "To do something, go to Y")
+
+Note: The format requirements are specifically defined in the memory initialization prompt and must be followed exactly.
+
+Input: {{input}}
+Output: {{output}}
+
+Does the output follow the exact required formatting structure with proper opening, #### headers, and bullet points? Be brutal
+- pass: Follows all required formatting specifications exactly
+- fail: Missing required formatting elements or includes prohibited content""",
+            choice_scores={
+                "pass": 1.0,
+                "fail": 0.0,
+            },
+            model="gpt-4.1",
+            **kwargs,
+        )
+
+
+@pytest.fixture(scope="module")
+def core_memory():
+    """We override the conftest.py core_memory fixture, which has autouse=True, as for memory onboarding a CoreMemory must be ABSENT."""
+    pass
+
+
+@pytest.fixture
+def call_memory_onboarding():
+    """Fixture to call MemoryOnboardingNode with a parametrized EventTaxonomyItem as input."""
+
+    async def callable(input: tuple[str, Optional[EventTaxonomyItem]]) -> Optional[AssistantMessage]:
+        organization_name, taxonomy_item = input  # The org name MAY matter as it gets included in the system prompt
+        # Because memory is stateful at the team level, each eval case here must have its own team instance
+        _, team, user = await database_sync_to_async(User.objects.bootstrap)(
+            organization_name=organization_name, email=f"{uuid4()}@example.com", password=None
+        )
+        conversation = await Conversation.objects.acreate(team=team, user=user)
+
+        graph = (
```

Comment:
As we use no checkpointer (and we use `DjangoCheckpointer` explicitly in other evals because of creating new event loops) - could it lead to flaky-ish behavior because of state loss?

Reply:
I don't fully get it either, so my assumption was that it controls the loops (also there's `# TRICKY: We need to set a checkpointer here because async tests create a new event loop.` comment).

Reply:
Made it so that we still have the mysterious checkpointer in. I think this one requires clarification from Georgiy, but it's okay both ways.